STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

akkinenic opened the pull request:
[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

bmeeder22 commented on org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java:
```diff
@@ -651,6 +657,48 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
+
+	public void format(List<? extends DiffEntry> entries, Pattern deltaFilterPattern) throws IOException {
+		if (deltaFilterPattern == null)
+			format(entries);
```
Also empty string ("")? also make a void return after this and we can eliminate the else block.

akkinenic replied:
deltaFilterPattern is an object. And if the pattern is null, it goes to the usual Jgit flow which is the format(entries).

bmeeder22 replied:
Ok, then just do the `return;` after so we can eliminate the else block.

akkinenic replied:
changed it

bmeeder22 commented on org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java:
```diff
@@ -651,6 +657,48 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
+
+				DiffEntry diffEntry = diIterator.next();
+
+				if (MODIFY.equals(diffEntry.changeType)) {
```
Don't need newlines between ALL of these lines... Makes the code harder to read.

akkinenic replied:
:)

akkinenic replied:
changed it

bmeeder22 commented on org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java:
```diff
@@ -651,6 +657,48 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
+						format(res.header, res.a, res.b);
+					} else {
+						diIterator.remove();
+					}
```
Break this into a private helper method.

akkinenic replied:
ah, I can do it. But its just 6 lines and it's already calling createFormatResult and  format methods.

bmeeder22 replied:
I get it, just makes it easier to read, makes your higher level function more clear in its actual intent.

akkinenic replied:
leaving in the same method

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

bmeeder22 commented on org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java:
```diff
@@ -651,6 +657,41 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
+				} else {
+					diIterator.remove();
+				}
+				} else {
```
Formatting issue on this line.

akkinenic replied:
fixed it

bmeeder22 reviewed (approved):
Fix the formatting otherwise good.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

axfiori commented on org.eclipse.jgit/src/org/eclipse/jgit/api/DiffCommand.java:
```diff
@@ -94,13 +96,17 @@
 
 	private ProgressMonitor monitor = NullProgressMonitor.INSTANCE;
 
+	private Pattern deltaFilterPattern = null;
```
can you add javadoc and mention that the filter is not going to be used if showNameAndStatusOnly is set to true

akkinenic replied:
added javadoc

axfiori commented on org.eclipse.jgit.test/tst/org/eclipse/jgit/api/DiffCommandTest.java:
```diff
@@ -241,6 +242,31 @@ public void testNoOutputStreamSet() throws Exception {
 
+	@Test
+	/**
+	 * Setting a delta filter of type ANY (applies to any file) with regex that matches the
```
can you update the description? it is from the time in which we had filter types. ty

akkinenic replied:
updated.

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.

wendts commented on org.eclipse.jgit.test/tst/org/eclipse/jgit/api/DiffCommandTest.java:
```diff
@@ -241,6 +242,31 @@ public void testNoOutputStreamSet() throws Exception {
+
+		String actual = out.toString();
+		String expected = "";
+		assertEquals(expected.toString(), actual);
```
Make these 3 a single line - assertEquals("", out.toString());

akkinenic replied:
fixed it

wendts commented on org.eclipse.jgit.test/tst/org/eclipse/jgit/api/DiffCommandTest.java:
```diff
@@ -241,6 +242,31 @@ public void testNoOutputStreamSet() throws Exception {
+		String actual = out.toString();
+		String expected = "";
+		assertEquals(expected.toString(), actual);
+	}
```
Should we replace all of the strings in this method (and following methods) with constants?

akkinenic replied:
Most of the format is the original Jgit tests format, intended to keep that way.

wendts commented on org.eclipse.jgit.test/tst/org/eclipse/jgit/diff/DiffFormatterTest.java:
```diff
@@ -343,6 +344,190 @@ public void testDiff() throws Exception {
+		String actual = os.toString("UTF-8");
+		String expected = "";
+
+		assertEquals(expected, actual);
```
Combine these - assertEquals("", os.toString("UTF-8"));

akkinenic replied:
changed it

wendts commented on org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java:
```diff
@@ -651,6 +657,41 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
+					format(res.header, res.a, res.b);
+				} else {
+					diIterator.remove();
+				}
```
No need for the curly braces around single lines

akkinenic replied:
oh yeah, I keep forgetting to not add those braces. style guide :(.

akkinenic replied:
changed it

Commit:
STRY50557163 : [Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

[Delta-Loading]Find the delta of changes between currently loaded application files and new changes applied while Import from SC

This is a customization in the JGIT to filter out the noise and reduce the source control app install on the service now instance.

The customization ONLY includes changes to DiffCommand and DiffFormatter and we exposed a new method and the purpose is to not break existing JGIT in any way.
